### PR TITLE
fix: catch filenotfound exception for rollover

### PR DESCRIPTION
**Pull request by bohan-amplitude**
### Summary

catch file not found exceptions for rollover

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

**Comment by yuhao900914** [comment]
Question:
This [issue](https://github.com/amplitude/Amplitude-Kotlin/issues/71) is related to `Too many file opens`.
If we catch the exception here and do nothing. Will we lost those events?

File: core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt [hunk 1]
```diff
@@ -165,10 +166,12 @@ class EventsFileManager(
...
-        FileOutputStream(file, true).use {
-            it.write(content)
-            it.flush()
-        }
+        try {
+            FileOutputStream(file, true).use {
+                it.write(content)
+                it.flush()
+            }
+        } catch (e: FileNotFoundException) {}
```
Comment (qingzhuozhen) [inline core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt:174]: When would this happen actually? Race condition for writing and deleting? Any thoughts on if we could recover from the issue? 

Another thought is for this kind of exception we could do some debug logging?

Comment (bohan-amplitude) [reply core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt:174]: I think file not found exceptions are happening when file content was sent out by another coroutine and deleted. This write operation are from rollover() -> finish() -> write a ']' to close the array in the file. Maybe try catch exception from the EventPipeline.update() where called storage.rollover() is better.

Comment (qingzhuozhen) [reply core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt:174]: OK, I guess as long as we could do a retry sounds good. Maybe better to mimic in a test case.

**Comment by bohan-amplitude** [comment]
@yuhao900914 The file not found exception happen when events already being processed and deleted by another thread. So we don't lose events. Not sure why `too many file opens` the exception seems can't find file by name. I'll clarify with user.

**Comment by liuyang1520** [comment]
Hi team, just a follow up on this fix, @yuhao900914 , @qingzhuozhen , is there any concerns on this fix?

**Comment by liuyang1520** [comment]
Rerun the test, still get the failure, @bohan-amplitude , could you pull the latest code from main? I didn't see the test failures on the recent PRs. Thanks!

File: core/src/main/java/com/amplitude/core/platform/EventPipeline.kt [hunk 2]
```diff
@@ -99,7 +99,13 @@ class EventPipeline(
         uploadChannel.consumeEach {
 
             withContext(amplitude.storageIODispatcher) {
-                storage.rollover()
+                try {
+                    storage.rollover()
+                } catch (e: FileNotFoundException) {
+                    e.message?.let {
```
Comment (qingzhuozhen) [inline core/src/main/java/com/amplitude/core/platform/EventPipeline.kt:105]: Do we need to return here? What happens for the later command reading from storage? Would it just recover?

**Comment by liuyang1520** [comment]
Open https://github.com/amplitude/Amplitude-Kotlin/pull/103 and close this one.

**Pull request closed without merging**